WrightTools/artists/_interact.py

```diff
@@ -0,0 +1,381 @@
+    return xaxis, yaxis
+
+
+def get_channel(data, channel):
+    if isinstance(channel, int):
+        channel = data.channels[channel]
+    elif isinstance(channel, str):
+        channel = [ch for ch in data.channels if ch.natural_name == channel][0]
+    elif type(channel) != wt_data.Channel:
+        raise TypeError("invalid channel type {0}".format(type(channel)))
+    return channel
+
+
+def get_colormap(channel):
+    if channel.signed:
+        cmap = "signed"
+    else:
+        cmap = "default"
+    cmap = colormaps[cmap]
+    cmap.set_bad([0.75] * 3, 1.)
+    cmap.set_under([0.75] * 3, 1.)
+    return cmap
+
+
+def get_clim(channel, local, local_arr=None):
+    if local:
+        if channel.signed:
+            mag = np.nanmax(np.abs(local_arr))
+            clim = [-mag, mag]
+        else:
+            clim = [0, np.nanmax(local_arr)]
+    else:
+        if channel.signed:
+            clim = [-channel.mag(), channel.mag()]
+        else:
+            clim = [0, channel.max()]
+    return clim
+
+
+def get_slices(sliders, axes, verbose=False):
+    slices = []
+    for axis in axes:
+        if axis.natural_name in sliders.keys():
+            ticklabels = gen_ticklabels(axis.points)
+            this_index = int(sliders[axis.natural_name].val)
+            sliders[axis.natural_name].valtext.set_text(ticklabels[this_index])
+            if verbose:
+                print(axis.natural_name, sliders[axis.natural_name].val, ticklabels[this_index])
+            slices.append(slice(this_index, this_index + 1))
+        else:
+            slices.append(slice(None))
+    return slices
+
+
+def gen_ticklabels(points):
+    step = np.diff(points).min()
+    if step == 0:
+        ticklabels = ['NaN' for point in points]
+        return ticklabels
+    ordinal = np.log10(np.abs(step))
+    ndigits = -int(np.floor(ordinal))
+    if ndigits < 0:
+        ndigits += 1
+        fmt = "{0:0.0f}"
+    else:
+        fmt = "{" + "0:.{0}f".format(ndigits) + "}"
+    ticklabels = [fmt.format(round(point, ndigits)) for point in points]
+    return ticklabels
+
+
+def norm(arr, signed, ignore_zero=True):
+    if signed:
+        norm = np.nanmax(np.abs(arr))
+    else:
+        norm = np.nanmax(arr)
+    if norm != 0 and ignore_zero:
+        arr /= norm
+    return arr
+
+
+def set_aspect(xaxis, yaxis):
+    if xaxis.units == yaxis.units:
+        xr = xaxis.max() - xaxis.min()
+        yr = yaxis.max() - yaxis.min()
+        aspect = np.abs(yr / xr)
+        if 3 < aspect or aspect < 1 / 3.:
+            raise Warning(
+                "units agree, but aspect {0} required for equal spacing is too"
+                / + "narrow.".format(aspect)
+            )
+            aspect = np.clip(aspect, 1 / 3., 3.)
+            print("using aspect {0} instead".format(aspect))
+    else:
+        aspect = 1
+    return aspect
+
+
+def interact2D(data, xaxis=0, yaxis=1, channel=0, local=False, verbose=True):
```

Comment:
Do we wish to allow `**kwargs` and pass those to the lower level plots?
Not sure we do, but want to consider it.

Reply:
I'm thinking things like colormaps and such (though the Qt viewer may provide some access as well)

Reply:
Perhaps we can consider this an enhancement to be added at a later date?